src/solvers/refinement/string_constraint_generator_main.cpp

```diff
@@ -409,13 +387,14 @@ array_string_exprt array_poolt::find(const exprt &pointer, const exprt &length)
 }
 
 /// Adds creates a new array if it does not already exists
-/// \todo This should be replaced by associate_char_array_to_char_pointer
+/// \todo This should be replaced
+/// by array_poolt.make_char_array_for_char_pointer
 array_string_exprt string_constraint_generatort::char_array_of_pointer(
   const exprt &pointer,
   const exprt &length)
 {
   const array_string_exprt array = array_pool.find(pointer, length);
-  add_default_axioms(array);
+  created_strings.insert(array);
   return array;
```

Comment:
This could be written as:

```c++
return *created_strings.insert(array_pool.find(pointer, length)).first;
```

`insert` and `emplace` return a pair containing an iterator to the newly inserted element and a boolean, which is set to true if the element was newly inserted. That way you can avoid creating a temporary and return a reference to the newly created element.

Reply:
(comment on commit message) creted -> created